### ISSUE TITLE
Steam Input Mousing Devices Result in Invisible Cursor on Wayland

### ISSUE THREAD

**Issue by pyro57** [ValveSoftware/steam-for-linux#9318]
#### Your system information

* Steam client version:  Mar 24 2023, at 17:20:43
* Distribution (e.g. Ubuntu):  Arch Linux
* Opted into Steam client beta?:  No
* Have you checked for system updates?:  Yes

#### Please describe your issue in as much detail as possible:
When Steam input loads in a Wayland session the cursor becomes invisible on the Steam Deck without an external mouse.  The mouse is visible during an X11 session, and before steam is launched the trackpads work as expected, after steam launches the cursor becomes invisible when using the built in controls on the steam deck.

External mice show a visible cursor, and the mouse is still clearly working using the trackpads and joysticks on the steam deck as things highlight and clicking does indeed work, but the cursor is invisible.  Its likely the same issue as this one [Issue](https://github.com/ValveSoftware/steam-for-linux/issues/6615) and has to do with steam input, but that issue was closed because of the ability to use 3rd party software to configure the controller instead of steaminput, which in my opinion is not a solution. 

TLDR: Steam input mice (trackpad and joystick on the steamdeck) make the cursor invisible on wayland sessions.  The cursor works fine until steam is launched then it becomes invisible. 

#### Steps for reproducing this issue:

1. start a wayland KDE session on the SteamDeck (or any pc running steam that has steaminput mousing devices, such as the Steam Controller).
2. see the cursor works correctly.
3. launch steam
4. see the cursor turn invisible when using the steam input mousing device.

**Comment by pyro57** [comment]
[Original issue](https://github.com/ValveSoftware/steam-for-linux/issues/6615)

**Comment by kisak-valve** [comment]
Hello @pyro57, this issue was previously evaluated multiple times and https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/431 probably has the most technical details behind it.

On the Steam side of this, this issue is already being tracked at #8604. Closing in favor of the older issue report.
